vllm/v1/structured_output/__init__.py

```diff
@@ -103,23 +131,36 @@ def grammar_bitmask(
                 self.backend.allocate_token_bitmask(
                     max_batch_size * (1 + max_num_spec_tokens))
 
+        bitmask_tensor = self._grammar_bitmask
```

Comment:
There's a lot of refactoring in here that does not seem necessary / related to thinking support